packages/cli/src/utils/helpers.ts

```diff
@@ -22,4 +24,42 @@ export async function getPubKeyFromAddrAndWeb3(addr: string, web3: Web3) {
   return pubKey
 }
 
+export async function nodeIsSynced(web3: Web3): Promise<boolean> {
+  try {
+    // isSyncing() returns an object describing sync progress if syncing is actively
+    // happening, and the boolean value `false` if not.
+    // However, `false` can also indicate the syncing hasn't started, so here we
+    // also need to check the latest block number
+    const syncProgress = await web3.eth.isSyncing()
+    if (typeof syncProgress === 'boolean' && !syncProgress) {
+      const latestBlock: Block = await web3.eth.getBlock('latest')
+      if (latestBlock && latestBlock.number > 0) {
+        // To catch the case in which syncing has happened in the past,
+        // has stopped, and hasn't started again, check for an old timestamp
+        // on the latest block
+        const ageOfBlock = Date.now() / 1000 - latestBlock.timestamp
```

Comment:
It might be good to extract this into a separate function.

Reply:
Do you mean something like `ageOfLatestBlock()` that would get the latest block and calculate its age? Or `ageOfBlock(block)` that would calculate the age of a block given as an argument?

Perhaps a more useful question:
What other use cases can you envision for such a helper function?